outputs.tf

```diff
@@ -36,3 +36,13 @@ output "existing_key_ring_keys" {
   description = "IDs of Keys created by the module in existing Key Rings"
   value       = module.existing_key_ring_keys
 }
+
+output "kp_private_endpoint" {
+  description = "Instance private endpoint URL"
+  value       = [for kp in module.key_protect : kp.kp_private_endpoint][0]
+}
+
+output "kp_public_endpoint" {
+  description = "Instance public endpoint URL"
+  value       = [for kp in module.key_protect : kp.kp_public_endpoint][0]
+}
```

Comment:
These outputs should be set to `null` if `existing_key_protect_instance_guid` is not null. So something like this:

`var.existing_key_protect_instance_guid == null ? [for kp in module.key_protect : kp.kp_public_endpoint][0] : null`

Reply:
Done.

Reply:
IMHO this should use `var.create_key_protect_instance` which is a boolean rather than an guid that might have been created in this terraform apply (in the root module, rather than in this module). Plus we know count is either 0 or 1, so a loop is not required.

```hcl
value = var.create_key_protect_instance ? module.key_protect[0].kp_public_endpoint : null
```

The syntax might vary if the kp_public_endpoints is a list it may require a [0] for the first element.

Reply:
good shout @shemau I had forgotten there was a boolean here